core/src/main/java/org/jruby/util/RubyFiletypeDetector.java

```diff
@@ -43,6 +43,9 @@ public String probeContentType(Path path) throws IOException {
             if (firstLine != null && SHEBANG_REGEXP.matcher(firstLine).matches()) {
                 return RUBY_MIME;
             }
+        } catch (Exception e) {
+            // Reading random files as UTF-8 could cause all sorts of errors
```

Comment:
Right, this is confusing. I guess it is for the case where the file does not exist, the caller might as well be warned.
But we can catch IOException anyway I think.
The current diff would also catch even a NullPointerException silently. This seems a good way to hide bugs.